pkg/kubelet/kubeletconfig/util/files/files_test.go

```diff
@@ -0,0 +1,296 @@
+	}{
+		{
+			name:         "not exists",
+			filePath:     "/tmp/FileExists",
+			needsCreated: false,
+			isFile:       false,
+			expectedBool: false,
+		},
+		{
+			name:         "exists as folder",
+			filePath:     "/tmp/FileExists",
+			needsCreated: true,
+			isFile:       false,
+			expectedBool: true,
+		},
+		{
+			name:         "exists as file",
+			filePath:     "/tmp/FileExists/file",
+			needsCreated: true,
+			isFile:       true,
+			expectedBool: true,
+		},
+	}
+
+	for _, testcase := range testcases {
+		var err error
+		t.Run(testcase.name, func(t *testing.T) {
+			if testcase.needsCreated {
+				if testcase.isFile {
+					// create as file
+					_, err = fs.TempFile(testcase.filePath, "")
+				} else {
+					// create as folder
+					err = fs.MkdirAll(testcase.filePath, defaultPerm)
+				}
+			}
+
+			if err != nil {
+				t.Errorf("unexpected error in creating file %s, err: %v", testcase.filePath, err)
+			} else {
+				if result, err := FileExists(fs, testcase.filePath); err != nil {
+					t.Errorf("unexpected error: %v", err)
+				} else {
+					if testcase.expectedBool != result {
+						t.Errorf("unexpected result, expected: %v, actual: %v", testcase.expectedBool, result)
+					}
+				}
+			}
+		})
+
+	}
+}
+
+func TestEnsureFile(t *testing.T) {
+	fs := utilfs.NewFakeFs()
+
+	testcases := []struct {
+		name     string
+		filePath string
+		ifExist  bool
+	}{
+		{
+			name:     "already exists",
+			filePath: "/tmp/EnsureFile",
+			ifExist:  true,
+		},
+		{
+			name:     "file not exists",
+			filePath: "/tmp/EnsureFile/test1",
+			ifExist:  false,
+		},
+	}
+
+	for _, testcase := range testcases {
+		var err error
+		t.Run(testcase.name, func(t *testing.T) {
+			if testcase.ifExist {
+				// should create the file before running the test
+				if _, err = fs.TempFile(testcase.filePath, ""); err != nil {
+					t.Fatalf("unexpeted error: %v", err)
+				}
+			}
+
+			if err = EnsureFile(fs, testcase.filePath); err != nil {
+				t.Errorf("unexpeted error in ensuring file: %v", err)
+			} else {
+				// verify if the file has been created
+				if _, err := fs.Stat(testcase.filePath); err != nil {
+					t.Errorf("unexpected error in verifying file: %s, error: %v", testcase.filePath, err)
+				}
+			}
+		})
+	}
+}
+
+func TestWriteTmpFile(t *testing.T) {
+	fs := utilfs.NewFakeFs()
+	path := "/tmp/WriteTmpFile/test"
+	fileContent := "a test case of WriteTmpFile"
+
+	// write content into file
+	filePath, err := WriteTmpFile(fs, path, []byte(fileContent))
+
+	if err != nil {
+		t.Errorf("unexpected error in WriteTmpFile: %v", err)
+	} else {
+		// read the content out and check if match the fileContent
+		result, err := fs.ReadFile(filePath)
+		if err != nil {
+			t.Errorf("unexpected error in reading written file: %v", err)
+		} else {
+			if fileContent != string(result) {
+				t.Errorf("unexpected content in written file, expected: %s, actual: %s", fileContent, result)
+			}
+		}
+	}
+}
+
```

Comment:
I haven't reviewed below this line, but you likely want to make some changes based on above comments.